src/main/java/http_server/ResponseMaker.java

```diff
@@ -0,0 +1,109 @@
+import java.nio.file.Paths;
+import java.util.List;
+
+import static http_server.StatusCodes.NOT_FOUND;
+import static http_server.StatusCodes.OK;
+import static java.util.Arrays.asList;
+
+public class ResponseMaker {
+
+    private RequestParser requestParser = new RequestParser();
+
+    public String statusResponse(String file) {
+        String statusCodeForResource = isResourceAvailable(file);
+
+        List<StatusCodes> statusCodes = asList(StatusCodes.values());
+        for (StatusCodes statusCode : statusCodes) {
+            if (statusCode.getStatusCode().equals(statusCodeForResource)) {
+                return buildStatusLine(statusCode);
+            }
+        }
+        return "Unhandled file type";
+    }
+
+
+    private boolean requestIsToHomePage(String resource) {
+        return resource.equals("/");
+    }
+
+    public String returnResourceContents(String resource) {
+        byte[] encodedContents;
+        String contents = null;
+
+        if (requestIsToHomePage(resource)) {
+            return "";
+        } else {
+            String filePath = String.format("public/%s", resource);
+            try {
+                encodedContents = Files.readAllBytes(Paths.get(filePath));
+                contents = new String(encodedContents, "US-ASCII");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return contents;
+        }
+    }
+
+    public String buildWholeResponse(String request) {
+        String typeOfRequest = requestParser.extractMethodFromRequest(request);
+        String resourceRequested = requestParser.parseResource(request);
+        if (isHeadRequest(typeOfRequest)) {
+            return returnNoMessageBody(resourceRequested);
+        } else {
+            return returnMessageBody(resourceRequested);
+        }
+    }
+
+    private String returnMessageBody(String resourceRequested) {
+        String fileContents = returnResourceContents(resourceRequested);
+        String contentType = returnContentType(requestParser.parseContentType(resourceRequested));
+        String response = "" + statusResponse(resourceRequested) + "\n" +
+                "Connection: close\n" +
+                String.format("Content-Type: %s\n\n", contentType);
+        return response + fileContents;
+    }
+
+    public String returnContentType(String fileExtension) {
+        switch (fileExtension) {
+            case "txt":
+                return "text/plain";
+            case "jpeg":
+                return "image/jpeg";
+            case "png":
+                return "image/png";
+            default:
+                return "image/gif";
+        }
+    }
+
+    private String returnNoMessageBody(String resourceRequested) {
+        String fileContents = "";
+        String response = fileContents + statusResponse(resourceRequested) + "\n";
+        return response + "\n";
+    }
+
+    private String isResourceAvailable(String resource) {
```

Comment:
Is this returning a boolean?  The name of the method implies that it should, but it doesn't look like it does.